src/snmalloc/mem/pool.h

```diff
@@ -141,16 +141,26 @@ namespace snmalloc
         }
       }
 
+      size_t request_size = bits::next_pow2(sizeof(T));
+      size_t round_sizeof = Aal::capptr_size_round(sizeof(T));
+      size_t spare = request_size - round_sizeof;
```

Comment:
Hm.  I'm not quite sure what the theorem for CHERI Concentrate looks like, exactly, but in general, `spare` bytes at `spare_start` is not guaranteed to be representable even if `request_size` and `round_sizeof` (both at `raw`) are.  But maybe that's OK, since we are just returning the `Range` to the cache, not actually constructing a capability with those bounds?

Presumably rounding up to powers of two is maximally convenient for the backend and, in addition, tends to give us a reasonable amount of `spare` space?  (If it got to be excessive, we could instead pad upwards by smaller powers of two, namely `__builtin_cheri_representable_alignment_mask(sizeof(T)) + 1`, and still not risk the backend trying to build a capability to a non-representable range.)

Reply:
The smallbuddy will split the returned amount into a collection of powers of two, so all nice and representable.